dhall-docs/README.md

```diff
@@ -97,6 +60,281 @@ dhall-docs --input . --output-link ${OTHER_LINK}
 
 For more information about the tool, check the `--help` flag.
 
+
+## Documenting your packages
+
+The markup language of the documentation is [CommonMark], which is a strict
+Markdown flavour. `dhall-docs` uses [`mmark`], a package that parses
+[CommonMark] and transforms it to HTML. This package follows (almost)
+this specification.
+
+Documenting your package is essentially writing comments on your source code,
+although there are some format rules that these needs to obey in order to work
+properly with `dhall-docs`. In a single file you can use any kind of comments.
+
+You can see examples of writing documentation on the [tests](./tasty/data/package)
+folder.
+
+### Block comments
+
+Normal block comments in Dhall starts with `{-` and ends with `-}`. For example:
+
+```dhall
+{- foo
+     bar
+       baz
+-}
+```
+
+A `dhall-docs` block comment needs to start with `{-|<NEW_LINE>`.
+That is, the actual documentation starts on the following line. The closing brace
+doesn't need any special treatment. This let you separate your own internal block
+comments from the ones that you want `dhall-docs` to output in the final markup.
+
+[CommonMark] supports some features that are sensitive to indentation (e.g.
+[indented-code-blocks]). On a `dhall-docs` block comment, the indentation is
+determined by the column of the `{` character from the opening brace. For
+example, on the following block comment:
+
+```dhall
+{-|
+foo
+  bar
+    baz
+-}
+```
+
+... and the following:
+
+```dhall
+              {-|
+              foo
+                bar
+                  baz
+                        -}
+```
+
+
+... will extract the contents of the comment, stripping the `{` leading whitespace
+from all other lines, returning the following text:
+
+```
+foo
+  bar
+    baz
+```
+
+The indentation level of a line is the number of whitespace characters (i.e. `' '`) from
+the first column until the column of the `{`. The rest of the whitespace that
```

Comment:
If tabs and space character are interchangeable, that means that e.g.

```
	{-|
 foo
 -}
```

is well-formed, right?

I don't know whether this could cause problems, but I think it might be worth spelling out.

Reply:
I think a lot of these issues would be fixed by using the exact same rules as multi-line strings in Dhall.  Then there's no confusion

Reply:
I think Gabriel has a point here. If we can specify how a doc comment corresponds to a multi-line string, we only need to add that the corresponding string is supposed to be the input for Markdown parser, and we're done with a good part of this spec.

I guess the tricky bit is that the multi-line spec isn't exactly super easy to understand itself.

How familiar are you with it at this stage, @german1608?

Reply:
@sjakobi: yes, that example is well formed.

@Gabriel439: I thought about sing something similar to multiline strings for this purpose, but I was thinking about something more strict for the comments and ended up using {-| as base of indentation.

Another idea that I had was to use the first line column as a base of indentation, that is more similar to multiline strings. But it will allow users to write things such as:

```
{-|
                  Foo
                      BAR
-}
```

That doesn't look to good IMO